src/routes/posts.js

```diff
@@ -58,6 +58,85 @@ const fetchPosts = async (req, res, next) => {
     }
 }
 
+// ASSIGN TAG TO AN ARTICLE
+
+const assignTagToPost = async (req, res) => {
+    try {
+        const { postId } = req.params // post id;
+        const { tagId, content, title } = req.body
+
+        const postsTags = await db.query(
+            'INSERT INTO posts_tags ("postId","tagId",content,title) SELECT $1,$2,$3,$4 WHERE NOT EXISTS (SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) RETURNING *',
+            [postId, tagId, content, title]
+        )
+
+        if (!postsTags.rows[0]) {
+            res.status(400).json({
+                message: 'Tag is already assigned to the post',
+            })
+        } else {
+            res.status(202).json({
+                message: 'Tag has been successfully assigned to the post',
+                tag: postsTags.rows[0],
```

Comment:
not actually sure if this is typically a `tag` in itself, but think for consistency with our other endpoints, it should just be `data` and let's intentionally return fields we want to return by choosing the specific values instead of sending back everything from the database

```suggestion
                data: {
                id: postsTags.rows[0].id // and so on
                },
```

Reply:
Resolved